### PR TITLE
tidying mod

### PR DESCRIPTION
**Pull request by drunkirishcoder**
some tidying up around the mods and how they are exported, only the new code is cleaned up

* the main change is to make sure each type is only exported externally through a single path. remove any ambiguity.  all mod paths should not be accessible externally are changed to private.
* reordered import lines top of the files slightly so `mod` come before `pub use` come before `use`.
* imports from internal paths now begin with `crate::` prefix so rustfmt will group them together. before they are intermingled with imports from external crates, confusing as to what are from dependent libs and what are internal.
* all tests mods are changed to private, any shared code (the sample buffers) are moved to their super scopes and gated by `#[cfg(test)]` separately.

### CODE REVIEW

File: framework/src/testing/mod.rs [hunk 1]
```diff
@@ -10,9 +10,9 @@
 //! netbricks = { version = "1.0", features = ["test"] }
 //! ```
 
-pub(crate) mod arbitrary;
```
Comment (zeeshanlakhani) [inline framework/src/testing/mod.rs]: ah I thought that these were needed to be used within nf-tests, no?

Comment (drunkirishcoder) [reply framework/src/testing/mod.rs]: yeah, all needed types are exported internally through `crate::testing` and externally through `netbricks::testing`. sub mods don't need any visibility.

**Comment by drunkirishcoder** [comment]
`crate::` is a new rust2018 feature. rest is just me being me.

**Comment by zeeshanlakhani** [comment]
@drunkirishcoder yeah, I mean, i'm ok w/ exposing public functions through submods in the way you're doing it (via self/wildcard). Not sure if that's typically kosher, but it does simply interfaces.

**Comment by drunkirishcoder** [comment]
oh yeah that's pretty standard rust. I see it a lot in the rust core libs. because the way rust does packaging based on files, that's the way to keep export paths sane for your crate while not having gigantic 10k line source files.

**Comment by zeeshanlakhani** [comment]
👍

**Pull request merged**